pkg/policy/condition/func.go

```diff
@@ -0,0 +1,168 @@
+/*
+ * Minio Cloud Storage, (C) 2018 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package condition
+
+import (
+	"encoding/json"
+	"fmt"
+	"sort"
+)
+
+// Function - condition function interface.
+type Function interface {
```

Comment:
Once you create an interface any user looking at the godoc will not be able to see why he/she needs to implement a said function which is missing. 

```
type Function interface {

    // String() - returns string representation of function.
    String() string
    // contains filtered or unexported methods
}
    Function - condition function interface.
```

This is confusing for an interface IMO @balamurugana we can simply export them.

Reply:
Basically I don't want to export `Function` type.  User needs to use `NewStringEqualsFunc()` and so on., but go tools throw warnings/errors about unexported name `function` would confuse user.  This is the primary reason here.

Exporting all methods in `Function` are not really useful and exposes many types/methods outside.

If such a requirement is in future, we can open when needed.